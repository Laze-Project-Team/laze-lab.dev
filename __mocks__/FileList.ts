class FileList<T> {
  _files: T[];
  length: number;
  constructor(files: T[]) {
    this._files = files;
    this.length = this._files.length;
  }
  item(index: number) {
    return this._files[index];
  }
  get [Symbol.toStringTag]() {
    return 'FileList';
  }
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.FileList = FileList;

export {};
