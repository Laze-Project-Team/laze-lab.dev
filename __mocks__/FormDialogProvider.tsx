import type { FC, ReactNode } from 'react';
import { useForm } from 'react-hook-form';

import { FormDialogContextProvider } from '@/components/ui/FormDialog/FormDialogContext';

export const FormDialogProvider: FC<{ children?: ReactNode }> = ({
  children,
}) => {
  const { control } = useForm();
  return (
    <FormDialogContextProvider
      value={{
        id: 'mock-form-dialog',
        control,
        onSubmit: async (data) => console.log(data),
      }}
    >
      {children}
    </FormDialogContextProvider>
  );
};
