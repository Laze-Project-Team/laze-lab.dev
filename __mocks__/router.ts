import type { useRouter } from 'next/router';
import { beforeEach, vi } from 'vitest';

export const mockRouter = (): void => {
  vi.mock('next/router', () => ({
    useRouter: () => {
      const router: ReturnType<typeof useRouter> = {
        asPath: '/',
        back: vi.fn(),
        basePath: '/',
        beforePopState: vi.fn(),
        events: {
          on: vi.fn(),
          off: vi.fn(),
          emit: vi.fn(),
        },
        isFallback: false,
        isLocaleDomain: false,
        isPreview: false,
        isReady: true,
        locale: 'en',
        locales: ['en'],
        prefetch: vi.fn(),
        push: vi.fn(),
        query: {},
        reload: vi.fn(),
        replace: vi.fn(),
        route: '/',
        pathname: '/',
      };

      return router;
    },
  }));
};

export const mockEachRouter = (): void => {
  beforeEach(() => {
    mockRouter();
  });
};
