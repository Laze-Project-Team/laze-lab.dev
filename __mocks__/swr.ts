import type { SWRResponse } from 'swr';

export const mockSWRResponse = <T>(data: T): SWRResponse<T> => ({
  isValidating: false,
  mutate: async () => void 0,
  data,
});
