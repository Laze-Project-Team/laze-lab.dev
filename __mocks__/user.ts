import type { User } from 'firebase/auth';

import type { userData } from '@/typings/database';

export const userMock: User | null | undefined = {
  uid: 'unique-id',
  delete: async () => void 0,
  displayName: 'Laze Project Team',
  email: 'laze@example.com',
  emailVerified: true,
  getIdToken: async () => 'token',
  getIdTokenResult: async () => ({
    authTime: new Date().toISOString(),
    expirationTime: new Date().toISOString(),
    issuedAtTime: new Date().toISOString(),
    signInProvider: null,
    signInSecondFactor: null,
    token: 'token',
    claims: {},
  }),
  isAnonymous: false,
  metadata: {},
  phoneNumber: null,
  photoURL: null,
  providerData: [],
  providerId: 'email',
  refreshToken: 'token',
  reload: async () => void 0,
  tenantId: null,
  toJSON: () => ({}),
};

export const userDataMock: userData | null | undefined = {
  name: 'Laze Project Team',
  avatarURL: 'https://laze.ddns.net/img/logo/logo.png',
  locale: 'ja',
  projects: [],
  createdAt: new Date(),
};
