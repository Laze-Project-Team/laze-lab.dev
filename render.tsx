import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
