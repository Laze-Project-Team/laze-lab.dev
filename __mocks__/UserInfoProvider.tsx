import type { FC, ReactNode } from 'react';
import { vi } from 'vitest';

import { mockSWRResponse } from '/__mocks__/swr';
import { userDataMock, userMock } from '/__mocks__/user';
import { userInfoContext } from '@/components/contexts/UserInfoContext';

export const syncUserMock = vi.fn();
export const syncUserDataMock = vi.fn();

export const UserInfoProvider: FC<{ children?: ReactNode }> = ({
  children,
}) => (
  <userInfoContext.Provider
    value={{
      user: mockSWRResponse(userMock),
      userData: mockSWRResponse(userDataMock),
      syncUser: syncUserMock,
      syncUserData: syncUserDataMock,
    }}
  >
    {children}
  </userInfoContext.Provider>
);
