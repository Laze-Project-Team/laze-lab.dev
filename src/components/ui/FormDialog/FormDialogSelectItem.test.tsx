import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';
import { expect, it } from 'vitest';

import { FormDialogProvider } from '/__mocks__/FormDialogProvider';
import { render } from '@/lib/test/render';

import { FormDialogSelectItem } from '.';

it('should be rendered correctly', () => {
  const { container } = render(
    <FormDialogSelectItem
      label="this is label"
      name="unique-name"
      options={[]}
    />,
    { wrapper: FormDialogProvider },
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('should have "button" role', () => {
  render(
    <FormDialogSelectItem
      label="this is label"
      name="unique-name"
      options={[]}
    />,
    {
      wrapper: FormDialogProvider,
    },
  );
  expect(screen.getByRole('button')).toBeInTheDocument();
});
