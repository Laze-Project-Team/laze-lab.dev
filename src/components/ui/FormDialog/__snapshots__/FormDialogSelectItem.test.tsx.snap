// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be rendered correctly 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-7xf5ru-MuiFormControl-root-MuiTextField-root-PresentialFormDialogSelectItem"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink="false"
    for="mock-form-dialog-unique-name"
    id="mock-form-dialog-unique-name-label"
  >
    this is label
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="mock-form-dialog-unique-name-label mock-form-dialog-unique-name"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mock-form-dialog-unique-name"
      role="button"
      tabindex="0"
    >
      <span
        class="notranslate"
      >
        â€‹
      </span>
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      defaultvalue=""
      name="unique-name"
      tabindex="-1"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-yjsfm1"
      >
        <span>
          this is label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
